hbase-server/src/test/java/org/apache/hadoop/hbase/master/TestSplitWALManager.java

```diff
@@ -86,6 +98,58 @@ public void teardown() throws Exception {
     TEST_UTIL.shutdownMiniCluster();
   }
 
+  @Test
+  public void testWALArchiveWithDifferentWalAndRootFS() throws Exception{
+    HBaseTestingUtility test_util_2 = new HBaseTestingUtility();
+    Path dir = TEST_UTIL.getDataTestDirOnTestFS("testWalDir");
+    test_util_2.getConfiguration().set(CommonFSUtils.HBASE_WAL_DIR, dir.toString());
+    CommonFSUtils.setWALRootDir(test_util_2.getConfiguration(), dir);
+    test_util_2.startMiniCluster(3);
+    HMaster master2 = test_util_2.getHBaseCluster().getMaster();
+    LOG.info("The Master FS is pointing to: " + master2.getMasterFileSystem()
+      .getFileSystem().getUri());
+    LOG.info("The WAL FS is pointing to: " + master2.getMasterFileSystem()
+      .getWALFileSystem().getUri());
+    Table table = test_util_2.createTable(TABLE_NAME, FAMILY);
+    test_util_2.waitTableAvailable(TABLE_NAME);
+    Admin admin = test_util_2.getAdmin();
+    MasterProcedureEnv env = test_util_2.getMiniHBaseCluster().getMaster()
+      .getMasterProcedureExecutor().getEnvironment();
+    final ProcedureExecutor<MasterProcedureEnv> executor = test_util_2.getMiniHBaseCluster()
+      .getMaster().getMasterProcedureExecutor();
+    List<RegionInfo> regionInfos = admin.getRegions(TABLE_NAME);
+    SplitTableRegionProcedure splitProcedure = new SplitTableRegionProcedure(
+      env, regionInfos.get(0), Bytes.toBytes("row5"));
+    // Populate some rows in the table
+    LOG.info("Beginning put data to the table: " + TABLE_NAME.toString());
+    int rowCount = 5;
+    for (int i = 0; i < rowCount; i++) {
+      byte[] row = Bytes.toBytes("row" + i);
+      Put put = new Put(row);
+      put.addColumn(FAMILY, FAMILY, FAMILY);
+      table.put(put);
+    }
+    executor.submitProcedure(splitProcedure);
+    LOG.info("Submitted SplitProcedure.");
+    test_util_2.waitFor(30000, () -> executor.getProcedures().stream()
```

Comment:
Shouldn't we filter by the split procedure submitted and assert it has completed success?